kdc/kerberos5.c

```diff
@@ -38,6 +38,44 @@
 #undef __attribute__
 #define __attribute__(X)
 
+struct kdc_pa_auth_state {
+	int auth_status;
+	const char *auth_details;
+	void *free_ptr;
+};
+
+static krb5_error_code
+_kdc_audit_auth_status(astgs_request_t r,
```

Comment:
Also, could this take a struct rather than individual arguments; you have the struct.

Reply:
Will push a fix shortly.